var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -73,6 +73,11 @@ class Gcc(AutotoolsPackage):
     variant('piclibs',
             default=False,
             description='Build PIC versions of libgfortran.a and libstdc++.a')
+    if sys.platform == 'darwin':
```

Comment:
i would keep this variant unconditional and instead add `conflicts('+rpathlibgcc', when=(sys.platform == 'darwin'))` or alike.